modules/serverless-creating-broker.adoc

```diff
@@ -3,41 +3,144 @@
-* Knative Eventing installed, which includes the `ClusterRole`.
+[source,yaml]
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+  namespace: default
+spec:
+  template:
+    spec:
+      containers:
+        - image: quay.io/openshift-knative/knative-eventing-sources-event-display
+----
 
-.Procedure
-. Create the `ServiceAccount` objects:
+.. Create the Service by applying the YAML file:
 +
+
+[source,terminal]
 ----
-$ oc -n default create serviceaccount eventing-broker-ingress
-$ oc -n default create serviceaccount eventing-broker-filter
+$ oc apply -f <FILENAME>
 ----
 
-. Give those objects RBAC permissions:
+. Optional. Create an event source that emits events to the `default` Broker. The following example shows how you can create a basic PingSource.
+.. Copy the following into a YAML file:
 +
+
+[source,yml]
 ----
-$ oc -n default create rolebinding eventing-broker-ingress \
-  --clusterrole=eventing-broker-ingress \
-  --serviceaccount=default:eventing-broker-ingress
-$ oc -n default create rolebinding eventing-broker-filter \
-  --clusterrole=eventing-broker-filter \
-  --serviceaccount=default:eventing-broker-filter
+apiVersion: sources.knative.dev/v1alpha2
+kind: PingSource
+metadata:
+  name: ping-source
+spec:
+  schedule: "*/1 * * * *"
+  jsonData: '{"message": "Hello serverless!"}'
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1beta1
+      kind: Broker
+      name: default
 ----
 
-. Create the broker:
 +
+..  Create the event source by applying the YAML file:
+
++
+[source,terminal]
+----
+$ oc apply -f <FILENAME>
+----
+
+. Create the Broker. You can do this by creating a Trigger object that uses the `knative-eventing-injection: enabled` annotation.
++
+[IMPORTANT]
+====
+If you create a Broker using the `knative-eventing-injection: enabled` annotation, you cannot delete this Broker without cluster administrator permissions.
+If you delete the Broker without having a cluster administrator remove this annotation first, the Broker will be created again after deletion.
+====
+
+.. Copy the following into a YAML file:
++
+
+[source,yml]
 ----
-cat << EOF | oc apply -f -
 apiVersion: eventing.knative.dev/v1beta1
-kind: Broker
+kind: Trigger
 metadata:
-  namespace: default
-  name: default <1>
-EOF
+  annotations:
+    knative-eventing-injection: enabled
+  name: example-trigger <1>
+spec:
+  broker: default
+  subscriber: <2>
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+----
+
++
+<1> You can specify any valid name for the Trigger.
+<2> The subscriber configuration is used to specify the event sink. In this example, it uses the Service that was created in step 1 of this procedure.
+
++
+..  Create the Broker by applying the YAML file:
++
+
+[source,terminal]
+----
+$ oc apply -f <FILENAME>
+----
+
+.Verification steps
+
+You can verify that events were sent to the event sink by looking at the sink pod's logs.
```

Comment:
Pod

So, if your pod isn't terminated before the next message is sent, do the verification steps work? I think that the intro needs a few changes, but I'm not sure what to suggest.

Reply:
cc @cardil who provided verification details.

Chris, what happens if the pod isn't terminated before the PingSource sends another message?
Maybe we need additional work on this section since it was never really that clear to me. I remember we had a bunch of Slack conversations and GitHub comments about what to add :slightly_smiling_face: 

I'd say if everything else in this PR is ready before this, we shouldn't block it on rewriting this since it's already there, but we should have a follow up issue / PR to clarify the verification steps ASAP.

Reply:
If pod isn't terminated before new event comes, the old one will receive it and display in its log. The command used in verification will still work just fine, though. Other examples in docs are using `*/2 * * * *` expression, which means that events will be sent with 2 minute interval. In this example we use `*/1 * * * *` expression, so it's 1 minute interval. Knative service has also 1 minute cool-down, by default, before termination. So in this case it's likely that the pod will not manage to wait enough to be terminated. It will be a race. Sometimes one, sometimes the other would happen. By using `*/2 * * * *` we can avoid that, but I personally wouldn't bother with it, as it shows different possibilities as well. And we are not focusing here on Serving functionality but Eventing.

I agree that we might upgrade verification steps for Eventing elements globally, in follow up PR. Also, upstream Knative has nicer command to display the logs of a pod. I rather use that one instead of what we have now.